app/routes.py

```diff
@@ -1,2 +1,187 @@
+goal_bp = Blueprint("goal", __name__, url_prefix="/goals")
+
+@task_bp.route("", methods=["GET"])
+def get_tasks():
+    all_tasks = Task.query.all()
+    if not all_tasks:
+        return jsonify([]), 200
+
+    sort_query = request.args.get("sort", default="asc") 
+    if sort_query == "desc":
+        all_tasks = Task.query.order_by(desc(Task.title)).all()
+    else:
+        all_tasks = Task.query.order_by(asc(Task.title)).all()
+
+    tasks_response = [task.convert_a_task_to_dict()for task in all_tasks]
+    return jsonify(tasks_response), 200
+
+@task_bp.route("", methods=["POST"])
+def post_a_task():
+    request_body = request.get_json()
+
+    try:  
+        if type(request_body["completed_at"]) == str:
+            datetime_completed_at = from_str_to_datetime(request_body["completed_at"])
+        else:
+            datetime_completed_at = request_body["completed_at"]
+
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=datetime_completed_at)
+        
+        db.session.add(new_task)
+        db.session.commit()
+
+        return new_task.concate_task_key_to_a_dict_with_return_code(201)
+    except KeyError:
+        return keyError_message()
+
+@task_bp.route("/<int:task_id>", methods=["GET"])
+def get_a_task_with_id(task_id):
+    task = Task.query.get_or_404(task_id)
+    return task.concate_task_key_to_a_dict_with_return_code(200)
+
+
+@task_bp.route("/<int:task_id>", methods=["PUT"])
+def put_a_task_with_id(task_id):
+    task = Task.query.get_or_404(task_id)
+    request_body = request.get_json()
+    try:
+        task.title = request_body['title']
+        task.description = request_body['description']
+
+        db.session.add(task)
+        db.session.commit()
+        return task.concate_task_key_to_a_dict_with_return_code()
+    except KeyError:
+        return keyError_message_require_all_fields()
+
+@task_bp.route("/<int:task_id>", methods=["DELETE"])
+def delete_a_task_with_id(task_id):
+    task = Task.query.get_or_404(task_id)
+    db.session.delete(task)
+    db.session.commit()
+    return {"details": f'Task {task.task_id} \"{task.title}\" successfully deleted'}
+
+@task_bp.route("<int:task_id>/<mark_complete_or_incomplete>", methods=["PATCH"])
+def mark_a_task_completed_with_id(task_id, mark_complete_or_incomplete):
+    task = Task.query.get_or_404(task_id)
+    if mark_complete_or_incomplete == "mark_complete": 
+        task.completed_at = date.today()
+
+        slack_channel_id = os.environ.get("SLACK_CHANNEL_ID")
+        slack_bot_token = os.environ.get('SLACK_BOT_TOKEN') #**
+
+        requests.post(
+        'https://slack.com/api/chat.postMessage',
+        headers={'Authorization': f'Bearer {slack_bot_token}'},
+        data={
+            'channel': slack_channel_id,      
+            'text': f'Someone just completed the task {task.title}'
+        }
```

Comment:
you could also move the slack functionality to a helper function or instance method for tasks to clean up this function